lambdas/issuecredential/src/main/java/uk/gov/di/ipv/cri/passport/issuecredential/IssueCredentialHandler.java

```diff
@@ -72,15 +97,44 @@ public APIGatewayProxyResponseEvent handleRequest(
             PassportCheckDao passportCheck =
                     dcsPassportCheckService.getDcsPassportCheck(resourceId);
 
-            PassportCriResponse passportCriResponse =
-                    PassportCriResponse.fromPassportCheckDao(passportCheck);
+            VerifiableCredential verifiableCredential =
+                    VerifiableCredential.fromPassportCheckDao(passportCheck);
 
-            return ApiGatewayResponseGenerator.proxyJsonResponse(
-                    HttpStatus.SC_OK, passportCriResponse);
+            SignedJWT signedJWT = generateAndSignVerifiableCredentialJwt(verifiableCredential);
+
+            return ApiGatewayResponseGenerator.proxyJoseResponse(
+                    HttpStatus.SC_OK, signedJWT.serialize());
         } catch (ParseException e) {
             LOGGER.error("Failed to parse access token");
             return ApiGatewayResponseGenerator.proxyJsonResponse(
                     e.getErrorObject().getHTTPStatusCode(), e.getErrorObject().toJSONObject());
+        } catch (JOSEException e) {
+            LOGGER.error("Failed to create VerifiableIdentityCredential");
+            return ApiGatewayResponseGenerator.proxyJsonResponse(
+                    HttpStatus.SC_INTERNAL_SERVER_ERROR, e.getLocalizedMessage());
```

Comment:
Not sure if this is the ideal kind of error we'd want to return. I'm not 100% sure as the error handling in general for CRI's is a bit inconsitant and hasn't fully been figured out. But so far I think we have either returned a Nimbus ErrorObject for proper Oauth errors, or one of our ErrorResponse objects

Reply:
I've fixed this to return an OAuth error status and message.